operator/rediscontroller/controller_test.go

```diff
@@ -0,0 +1,120 @@
+package rediscontroller
```

Comment:
This is a preliminary setup of how we could unit test the controllers. I'm not sure how much worth they are but my goal is to have a middleground between e2e tests and manual tests. This way we could test `Observe()`, `Create()`, `Update()`, and `Delete()` better without the need to have a complete crossplane setup. 

Of course the whole `httptest.Server` thing could be improved a lot, this is just a start and a RFC.

Reply:
That's a good idea to start to have some kind of unit tests in the main methods. We generally skipped this for quite some time but we planned in the future to add such tests.

Reply:
envtest: yeah saw it and didn't see it in use so I didn't use it :) But yeah - that's fine to me, I'll refactor this accordingly :+1: 

I'll add a few tests for redis then and we can reuse the pattern/suite for other tests in the future.